Source/Core/Core/Brawlback/BrawlbackUtility.cpp

```diff
@@ -5,15 +5,72 @@
 
 namespace Brawlback
 {
+    void ResetRollbackInfo(RollbackInfo& rollbackInfo)
+    {
+      rollbackInfo.isUsingPredictedInputs = false;
+      rollbackInfo.beginFrame = 0;
+      rollbackInfo.endFrame = 0;
+      rollbackInfo.predictedInputs = FrameData();
+      rollbackInfo.pastFrameDataPopulated = false;
+      memset(rollbackInfo.pastFrameDatas, 0, sizeof(FrameData) * MAX_ROLLBACK_FRAMES);
+      rollbackInfo.hasPreserveBlocks = false;
+    }
 
     bool isButtonPressed(u16 buttonBits, PADButtonBits button)
     {
         return (buttonBits & (PADButtonBits::Z << 8)) != 0;
     }
+    /*
+     * Simple checksum function stolen from wikipedia:
+     *
+     *   http://en.wikipedia.org/wiki/Fletcher%27s_checksum
+     */
+
+    int fletcher32_checksum(short* data, size_t len)
+    {
+      int sum1 = 0xffff, sum2 = 0xffff;
+
+      while (len)
+      {
+        size_t tlen = len > 360 ? 360 : len;
+        len -= tlen;
+        do
+        {
+          sum1 += *data++;
+          sum2 += sum1;
+        } while (--tlen);
+        sum1 = (sum1 & 0xffff) + (sum1 >> 16);
+        sum2 = (sum2 & 0xffff) + (sum2 >> 16);
+      }
+
+      /* Second reduction step to reduce sums to 16 bits */
+      sum1 = (sum1 & 0xffff) + (sum1 >> 16);
+      sum2 = (sum2 & 0xffff) + (sum2 >> 16);
+      return sum2 << 16 | sum1;
+    }
+
+    // super slow
+    int SavestateChecksum(std::vector<ssBackupLoc>* backupLocs)
+    {
+      if (backupLocs->empty())
+        return -1;
+      std::vector<int> sums = {};
+      for (ssBackupLoc backupLoc : *backupLocs)
+      {
+        short* data = (short*)Memory::GetPointer(backupLoc.startAddress);
+        size_t size = backupLoc.endAddress - backupLoc.startAddress;
+        if (data && size)
+          sums.push_back(fletcher32_checksum(data, backupLoc.endAddress - backupLoc.startAddress));
+        else
+          ERROR_LOG(BRAWLBACK, "Invalid data or size of savestate when computing checksum!\n");
+      }
+      return fletcher32_checksum((short*)sums.data(), sizeof(int) * sums.size());
+    }
 
-    Match::PlayerFrameDataImpl* findInPlayerFrameDataQueue(const PlayerFrameDataQueue& queue, u32 frame) {
-        Match::PlayerFrameDataImpl* ret = nullptr;
-        #if 0
+    PlayerFrameData* findInPlayerFrameDataQueue(const PlayerFrameDataQueue& queue, u32 frame)
+    {
+      PlayerFrameData* ret = nullptr;
+#if 0
```

Comment:
Not a fan of code hanging around behind a #if 0 like this.

Is this needs to be kept, at least use `#ifdef SOME_NAME` here instead and document somewhere how / when to use these switches

Reply:
This is OOS for this PR. @FaultyPine Any way you can go in and fix after?